src/common/snippets/src/lowered/pass/serialize_data_flow.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/serialize_data_flow.hpp"
+
+#include "openvino/pass/serialize.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/op/serialization_node.hpp"
+#include "snippets/snippets_isa.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+bool SerializeDataFlow::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SerializeDataFlow")
+    if (linear_ir.empty())
+        return false;
+
+    ov::ResultVector results;
+    ov::ParameterVector parameters;
+    std::map<ExpressionPtr, std::shared_ptr<Node>> ops_map;
+    for (const auto& expr : linear_ir) {
+        const auto node = expr->get_node();
+        ov::OutputVector inputs(expr->get_input_count());
+        for (size_t i = 0; i < expr->get_input_count(); ++i) {
+            const auto& input_expr = expr->get_input_port_connector(i)->get_source().get_expr();
+            OPENVINO_ASSERT(ops_map.count(input_expr), "input node wasn't found during serialization");
+            inputs[i] = ops_map[input_expr]->output(0);
```

Comment:
Should we take output by source port here?
```suggestion
            inputs[i] = ops_map[input_expr]->output(expr->get_input_port_connector(i)->get_source().get_index());
```
At the moment Snippets support many ops with one output. This is the most common case.. But, for example, `BrgemmCopyB` has 2 outputs. I think it may be broken here.

Reply:
Since `SerializationNode` always has only one output, LIR with `BrgemmCopyB` with 2 outputs is serialized successfully (its children are connected to a single output). However, your point is valid: we lose some information about data flow in case of ops with several outputs.
Let's discuss possible solutions and criticality of the issue offline

Reply:
I applied you suggestion for data flow serialization pass. However, in control flow serialization, all nodes must have single output (since in this case model represents linear execution flow). To separate these 2 cases, I added a filed `SerializationMode` to serialization mode